controllers/managedocs_controller.go

```diff
@@ -1086,3 +1118,66 @@ func getCSVByPrefix(csvList opv1a1.ClusterServiceVersionList, name string) *opv1
 	}
 	return csv
 }
+
+func (r *ManagedOCSReconciler) reconcileK8sMetricsServiceMonitorAuthSecret() error {
+	r.Log.Info("Reconciling k8s-metrics-service-monitor-auth secret")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.k8sMetricsServiceMonitorAuthSecret, func() error {
+		secret := &corev1.Secret{}
+		secret.Name = grafanaDatasourceSecretName
+		secret.Namespace = openshiftMonitoringNamespace
+		if err := r.unrestrictedGet(secret); err != nil {
+			return fmt.Errorf("Failed to get grafana-datasources secret from openshift-monitoring namespace: %v", err)
+		}
+
+		authInfoStructure := struct {
+			DataSources []struct {
+				BasicAuthPassword string `json:"basicAuthPassword"`
+				BasicAuthUser     string `json:"basicAuthUser"`
+			} `json:"datasources"`
+		}{}
+
+		grafanaDatasourcesPrometheusConfig := secret.Data[grafanaDatasourceSecretKey]
+
+		if err := json.Unmarshal(grafanaDatasourcesPrometheusConfig, &authInfoStructure); err != nil {
+			return err
+		}
+		if err := r.own(r.k8sMetricsServiceMonitorAuthSecret); err != nil {
+			return err
+		}
+
+		r.k8sMetricsServiceMonitorAuthSecret.Data = map[string][]byte{
+			"Username": []byte("internal"),
+		}
+		for _, value := range authInfoStructure.DataSources {
+			if value.BasicAuthUser == "internal" {
+				if value.BasicAuthPassword == "" {
+					return fmt.Errorf("k8s-metrics-service-monitor-auth secret does not contain a Password entry")
+				}
+				r.k8sMetricsServiceMonitorAuthSecret.Data["Password"] = []byte(value.BasicAuthPassword)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to create k8s-metrics-service-monitor-auth secret: %v", err)
+	}
+	return nil
+}
+
+func (r *ManagedOCSReconciler) reconcileK8sMetricsServiceMonitor() error {
+	r.Log.Info("Reconciling k8s-metrics-service-monitor Service Monitor")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.k8sMetricsServiceMonitor, func() error {
+		if err := r.own(r.k8sMetricsServiceMonitor); err != nil {
+			return err
+		}
+		desired := templates.K8sMetricsServiceMonitorTemplate.DeepCopy()
+		r.k8sMetricsServiceMonitor.Spec = desired.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to create k8s-metrics-service-monitor service monitor: %v", err)
+	}
```

Comment:
Same as the last comment regarding resource names, also it will rarely be a create, most of the time it will be an update. 

```suggestion
	if err != nil {
		return fmt.Errorf("Failed to update k8sMetricsServiceMonitor: %v", err)
	}
```